### PR TITLE
Define handling of multiple applicable modifiers

### LINKED ISSUE #684: Define handling of multiple applicable modifiers
1) From what I can tell, the spec doesn't currently define which `additionalDisplayItems` should be shown if a UA decides to show any.
2) More generally, handling of multiple applicable modifiers isn't clear. It looks like it's possible to have multiple `methodData` and `modifiers` with the same `supportedMethods` including the same `data`.

Consider the following example snippet:
```js
// Credit card incurs a $3.00 processing fee.
const creditCardFee = {
  label: "Credit card processing fee",
  amount: { currency: "USD", value: "3.00" },
};

// Visa card incurs a $1.00 processing fee.
const visaFee = {
  label: "Visa processing fee",
  amount: { currency: "USD", value: "1.00" },
};

// Modifiers apply when the user chooses to pay with
// a credit card.
const modifiers = [
  {
    additionalDisplayItems: [creditCardFee],
    supportedMethods: "basic-card",
    total: {
      label: "Total due",
      amount: { currency: "USD", value: "53.00" },
    },
    data: {
      supportedTypes: ["credit"],
    },
  },
  {
    additionalDisplayItems: [visaFee],
    supportedMethods: "basic-card",
    total: {
      label: "Total due",
      amount: { currency: "USD", value: "54.00" },
    },
    data: {
      supportedNetworks: ["visa"]
    },
  },
];
Object.assign(details, { modifiers });
```
If multiple modifiers match the selected payment method (e.g. if the user selects a Visa card):
* Should all applicable additional display items be shown? The first matching one? Last matching one? Most specific one (would need to be defined)
* Which modifier's `total` should be used?

This seems like something that should be tested as well.

### LINKED ISSUE #753: Modifiers and [[serializedModifierData]] not actually used anywhere.
Following on from #684, there are a bunch of huge issues in the spec relating to modifiers:

 * [[serializedModifierData]] is not actually used anywhere - it's undefined how that data actually gets applied/displayed.
 * [[serializedModifierData]] gathers all the data from EVERY modifier (without matching on PMI!), then passes ALL data to the chosen payment handler - in the spec, there is no way to distinguish between data for ApplePay and data for Basic-Card.  
 * The modifiers in `serializedModifierData` are never converted, only serialized, meaning that if they are  converted by the payment handler, they can throw: This is undefined behavior.  
 * The actual serialization also seems pointless. If that's for IPC, then that should be an implementation detail. It doesn't seem to serve any purpose in the spec (because [[serializedModifierData]] is never actually used for anything). 

I'm again strongly inclined towards dropping modifiers in favor of something else. They are super broken, underspecified, and unnecessarily complicated.

### PR DESCRIPTION
**Pull request by marcoscaceres**
closes #684 
closes #753 

The following tasks have been completed:

 * [x] Confirmed there are no ReSpec errors/warnings.
 * [ ] Modified Web platform tests - will test in Basic Card. 
 * [ ] Modified MDN Docs (link)
 * [ ] Has undergone security/privacy review (link)
 
Implementation commitment:

Only applies at the Payment Handler level. This PR just gives guidance.  

Optional, impact on Payment Handler spec?

Payment Handler spec needs to probably say something about matching order.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/pull/824.html" title="Last updated on Jan 31, 2019, 2:37 AM UTC (e4352a4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/824/5966e82...e4352a4.html" title="Last updated on Jan 31, 2019, 2:37 AM UTC (e4352a4)">Diff</a>

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@ianbjacobs, could you give this first pass review? 🎁

File: index.html [hunk 1]
```diff
@@ -1055,19 +1057,104 @@ <h2>
...
+    total: {
+      label: "Total due",
+      amount: { currency: "USD", value: "51.00" },
+    },
+    data: {
+      supportedNetworks: ["visa"] // Visa network only
+    },
+  },
+];
+Object.assign(details, { modifiers });
```
Comment (domenic) [inline index.html]: Not sure where `details` is declared, but just `details.modifiers = modifiers` is simpler.

File: index.html [hunk 2]
```diff
@@ -1055,19 +1057,104 @@ <h2>
...
+              </pre>
+              <p>
+                If the modifiers array in the example above was to be reversed
+                (i.e., <code class="js">supportedNetworks: []</code>) was to
```
Comment (domenic) [inline index.html]:
```suggestion
                (i.e., <code class="js">supportedNetworks: []</code> was to
```

**Review by domenic — APPROVED** [review]
LGTM with nits

**Pull request merged**